### ISSUE TITLE
Test TestSumLargeDynRange not passing for CPU

### ISSUE THREAD

**Issue by nikolovjovan** [riakymch/exblas#2]
TLDR:
After resolving troubles with compiling this project, I have tried running tests which resulted in TestSumLargeDynRange test failing.

OS: Pop!_OS 21.10
Kernel: 5.13.0-7614-generic

First I tried using latest gcc (11.2.0) with Intel OneAPI TBB 2021.4.0. This failed miserably with many errors. First narrowing conversions resulted in template for constant8f not being generated in certain instances. I "fixed" that by updating vectorclass to 2.0. However, tbb was still old so I had to fix task_scheduler_init. After fixing all that, include and lib dirs I finally built the project.

However, after running tests I realized TestSumLargeDynRange fails.

Since I have changed a bunch of code only to build the project, I decided to try to use an older version of gcc (8.5.0) with older tbb version (before OneAPI - TBB 4.4 Update 6). After making this work by changing include and lib paths this test still fails.

Do you know if tests are broken/lib is broken or is there an issue on my end? I would like to use this project in my master's thesis so I would like to know it works before attempting to use it.

Thanks in advance!

**Comment by nikolovjovan** [comment]
Forgot to add only compiled for CPU so far. Tried compiling for GPU (NVIDIA) but program had troubles finding kernel and I did not look much deeper into it.
